### ISSUE TITLE
The data pipeline testing module of the TESTING_TOOL must be able to inject additional pipeline components into the data pipeline for generating synthetic input data

### ISSUE THREAD

**Issue by avanhoorn** [radon-h2020/radon-ctt#32]
From D2.1 (Companion)

ID | R-T3.3-3
------------ | -------------
Section | WP3: Methodology and Quality Assurance Requirements
Type| FUNCTIONAL_SUITABILITY
User Story | As a QoS Engineer, I want to be able to inject data producing components into the data pipeline (under testing) for generating test data into the pipeline.
Requirement | The data pipeline testing module of the TESTING_TOOL must be able to inject additional pipeline components into the data pipeline for generating synthetic input data.
Extended Description | The goal is to support creating and adding data pipeline components that mock the input streams by generating synthetic data into the data pipeline.
Priority | Must have
Affected Tools | TESTING_TOOL
Means of Verification | Tests with different RADON data pipelines to verify that the components were injected and the modified data pipeline can be deployed successfully
Dependency | N/A

Superseded by #83

**Comment by avanhoorn** [comment]
@pjakovits Could you set the milestone, please?

**Comment by pjakovits** [comment]
> @pjakovits Could you set the milestone, please?

Please assign  milestone M27 to this issue @avanhoorn

**Comment by pjakovits** [comment]
This issue should be closed as the requirement was changed to match better with the new modular and agent-based architecture of the CTT tool. We identified that injecting components into the data pipeline under test is no longer required to fulfill the original user requirement and a data generation agent is a better solution. References to the specific approach to address the requirement were removed.  
New issue: #83
